src/main/navigation/navigation_pos_estimator.c

```diff
@@ -405,7 +412,19 @@ static void updateIMUTopic(timeUs_t currentTimeUs)
         /* Read acceleration data in NEU frame from IMU */
         posEstimator.imu.accelNEU.x = accelBF.x;
         posEstimator.imu.accelNEU.y = accelBF.y;
-        posEstimator.imu.accelNEU.z = accelBF.z;
+        const float baroTemperature = baroGetTemperature();
+        /* Some accelerometers may experience drifting caused by changing temperatures, 
+        you can use these 2 parameters to establish the linear relationship between the accelerometer drifting and the temperature 
+        (the baro needs to be in the FC, where the accelerometer is located)
+        The formula is the following: f(x) = ax + b
+        where f(x) is the calculated offset, a is how the offset varies for each degree, x is the barometer temperature in celsius and b would be the offset for 0ºC scenario 
+         What works well for me is:
+        temp_correction_a = 4.834
+        temp_correction_b = -217.53
+         */
+        const float acc_z_offset = ( positionEstimationConfig()->temp_correction_a * ( baroTemperature / 10.0f ) ) + positionEstimationConfig()->temp_correction_b; // f(x) = ax + b
```

Comment:
You should be using the accelerometer temps, not the baro temps for this. The baro could be very far from the accelerometer and not be exposed to the same local temperature gradients.